scripts/tools/RunConfigurationMarkers.php

```diff
@@ -0,0 +1,177 @@
+ *
+ * Copyright (c) 2023 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace oat\tao\scripts\tools;
+
+use common_ext_ExtensionsManager;
+use Laminas\ServiceManager\ServiceLocatorAwareTrait;
+use oat\oatbox\extension\script\ScriptAction;
+use oat\oatbox\reporting\Report;
+use oat\tao\model\configurationMarkers\ConfigurationMarkers;
+use oat\oatbox\service\ServiceManager;
+use oat\generis\model\DependencyInjection\ContainerStarter;
+
+/**
+ * Usage
+ * php index.php 'oat\tao\scripts\tools\RunConfigurationMarkers' -s path/to/seed.json
+ * php index.php 'oat\tao\scripts\tools\RunConfigurationMarkers' -s path/to/seed.json -e generis
+ */
+class RunConfigurationMarkers extends ScriptAction
+{
+    use ServiceLocatorAwareTrait;
+
+    private const OPTION_SELECT_EXTENSION_ID = 'select-extension-id';
+    private const OPTION_SEED_FILE_PATH = 'seed-path';
+    private Report $report;
+
+    protected function provideOptions(): array
+    {
+        return [
+            self::OPTION_SELECT_EXTENSION_ID => [
+                'prefix' => 'e',
+                'longPrefix' => self::OPTION_SELECT_EXTENSION_ID,
+                'description' => 'Select single extensions config to change.',
+            ],
+            self::OPTION_SEED_FILE_PATH => [
+                'prefix' => 's',
+                'longPrefix' => self::OPTION_SEED_FILE_PATH,
+                'description' => 'Path to seed file to use.',
+            ],
+        ];
+    }
+
+    protected function provideDescription(): string
+    {
+        return 'Remove secret data from extension\'s config file using markers in seed file';
+    }
+
+    protected function run(): Report
+    {
+        $this->report = Report::createInfo('Starting.');
+        if ($this->hasOption(self::OPTION_SEED_FILE_PATH) === false) {
+            return Report::createError(sprintf('Option %s is mandatory.', self::OPTION_SEED_FILE_PATH));
+        }
+
+        $filePath = $this->getOption(self::OPTION_SEED_FILE_PATH);
+        $fileContents = file_get_contents($filePath);
+        if ($fileContents === false) {
+            $this->report->add(Report::createError('Empty seed file or wrong file path, aborting.'));
+
+            return $this->report;
+        }
+        $parameters = json_decode($fileContents, true);
+        if (json_last_error() !== JSON_ERROR_NONE) {
+            $this->logError(
+                sprintf(
+                    'Json file malformed, last json error : %s , message: %s',
+                    json_last_error(),
+                    json_last_error_msg()
+                )
+            );
+            $this->report->add(
+                Report::createError('Json file contains errors see logs for details, aborting.')
+            );
+
+            return $this->report;
+        }
+
+        $container = $this->getServiceLocator()->getContainer();
+        $markers = $container->get(ConfigurationMarkers::class);
```

Comment:
```suggestion
        $container = $this->getServiceLocator()->getContainer();
        $markers = $container->get(ConfigurationMarkers::class);
```

Could you export it to method `getConfigurationMarkers` it will return ConfigurationMarkers class.

Reply:
Done